src/_pytest/assertion/rewrite.py

```diff
@@ -273,13 +273,15 @@ def get_data(self, pathname: Union[str, bytes]) -> bytes:
         with open(pathname, "rb") as f:
             return f.read()
 
-    if sys.version_info >= (3, 9):
+    if sys.version_info >= (3, 10):
 
         def get_resource_reader(self, name: str) -> importlib.abc.TraversableResources:  # type: ignore
```

Comment:
Hey, I'm currently writing a code analysis tool for Python projects. It pointed me to this code part, which I find quite interesting. Hence, I would have some questions about it:

1. Where exactly is the `get_resource_reader()` function called? I can't find any references to it. 
2. Why is the function defined by a condition? Would it make sense to move the version check inside the function with an early return? (As there is already a version check within it) As the function is public, it would keep the interface of the `AssertionRewritingHook` independent to the Python version.

I think these questions may be adressed to @kdelee or @nicoddemus? I would be really happy to hear from you as this would give me some qualitative insights on the reports of my tool.

Kind regards,
Matthias

Reply:
it's called by the python import machinery itself. this function isn't valid until 3.10 (it breaks 3.9)

Reply:
I saw the Github issue on the FileReader import causing the break for 3.9

Before, the import statement in the function was without any version check. But having the check `sys.version_info >= (3, 10)` within the function with an early return shouldn't break anything as the import statement would never be reached for 3.9. 
Or am I overlooking anything why the function would be invalid otherwise?

Reply:
there's backports of parts of importlib which would reach into this and if the function is defined try and call it (and then error) -- by not defining the function, those backports won't even attempt to call it and therefore won't crash

Reply:
Ahh ok I understand. Thanky you very much for your explanation, this helps me a lot!